translate/storage/mozilla_lang.py

```diff
@@ -121,22 +136,24 @@ def parse(self, lines):
 
             if line.startswith(';'):
                 source_unit = self.addsourceunit(line[1:])
+                source_unit.eol = self.eol
                 source_unit.addlocation(
                     "%s:%d" % (self.filename, lineoffset + 1))
                 if comment is not None:
                     source_unit.addnote(comment[:-1], 'developer')
                     comment = ""
 
     def serialize(self, out):
+        eol = self.eol.encode('utf-8')
         if self.is_active or self.mark_active:
-            out.write(b"## active ##\n")
+            out.write(b"## active ##")
+            out.write(byte(self.eol))
```

Comment:
We need self.eol to be unicode throughout and then just bytes at the end.  So no we don't want to clutter things elsewhere.